include/boost/variant/recursive_wrapper.hpp

```diff
@@ -16,6 +16,7 @@
 #include <boost/variant/recursive_wrapper_fwd.hpp>
 #include <boost/variant/detail/move.hpp>
 #include <boost/checked_delete.hpp>
+#include <boost/core/exchange.hpp>
```

Comment:
Add `BOOST_ASSERT` for all the cases when recursive_wrapper is dereferenced.

Reply:
Mostly for all the `apply_visitor`s

Reply:
Fixed by adding assertation to `recursive_wrapper::get` method which is used in every apply_visitor.